kolibri/core/content/api.py

```diff
@@ -1310,6 +1315,79 @@ def consolidate(self, items, queryset):
         return sorted_items
 
 
+class ContentDownloadRequestViewset(ValuesViewset):
+    pagination_class = OptionalPageNumberPagination
+
+    queryset = ContentDownloadRequest.objects.all()
+
+    field_map = {
+        "source_id": "self.request.user.id",
+    }
+
+    values = (
+        "id",
+        "requested_at",
+        "reason",
+        "contentnode_id",
+        "metadata",
+        "status",
+    )
+
+    def filter_queryset(self, queryset):
+        return queryset.filter(source_id=self.request.user.id)
+
+    def annotate_queryset(self, queryset):
+        return queryset.annotate(
+            has_removal=Exists(
+                ContentRemovalRequest.objects.filter(
+                    source_model=OuterRef("source_model"),
+                    source_id=OuterRef("source_id"),
+                    contentnode_id=OuterRef("contentnode_id"),
+                    requested_at__gte=OuterRef("requested_at"),
+                )
+            )
+        ).filter(has_removal=False)
+
+    def create(self, request, *args, **kwargs):
+        # if there is an existing deletion request, delete the deletion request
+        deletion_request = ContentRemovalRequest.objects.filter(
+            contentnode_id=self.request["contentnode_id"],
+            source_id=self.request.user.id,
+        )
+
+        if deletion_request:
+            ContentRemovalRequest.objects.filter(
+                contentnode_id=self.request["contentnode_id"],
+                source_id=self.request.user.id,
+            ).delete()
+        else:
+            try:
+                request = ContentDownloadRequest.build_for_user(self, *args)
+                # this is definitely not right but I don't know what would be? but I need to put that somewhere, I think
+                # i.e.
+                metadata = ContentNode.objects.filter(
+                    pk=OuterRef("contentnode_id")
+                ).metadata
+                ContentRequest.metadata = metadata
+                return request
```

Comment:
The `create` and `delete` methods should return a Django REST framework response object, see here for an example of a `create` method doing this: https://github.com/learningequality/kolibri/blob/a1afc9594526aff72d587f221783f332eaa2d539/kolibri/core/device/api.py#L103

The status code as a 201, I think we can do regardless of whether we created it or not.